R/submit-model.R

```diff
@@ -200,3 +203,272 @@ submit_nonmem_model <- function(.mod,
+  } else if (.model_type == "stan") {
+    stop(NO_STAN_ERR_MSG)
+  } else {
+    stop(glue("Passed `{.model_type}`. Valid options: `{SUPPORTED_MOD_TYPES}`"))
+  }
+  return(res_list)
+}
+
+
+#' S3 dispatch for submit_models from numeric input
+#' This will only work if you are calling from the same directory as the models, or if you have set the model directory with `set_model_directory()`
+#' @param .mods Integer vector that corresponds to the names of YAML and model files
+#' @param .directory Model directory containing the files referenced by `.mods`. Defaults to `options('rbabylon.model_directory')`, which can be set globally with `set_model_directory()`.
+#' @export
+#' @rdname submit_models
+submit_models.numeric <- function(
+  .mods,
+  .bbi_args = NULL,
+  .mode = c("sge", "local"),
+  ...,
+  .config_path=NULL,
+  .wait = TRUE,
+  .dry_run=FALSE,
+  .directory = getOption("rbabylon.model_directory")
+) {
+  # convert to character
+  .mods <- as.character(.mods)
+
+  # call character dispatch
+  res_list <- submit_models(
+    .mod = .mods,
+    .bbi_args = .bbi_args,
+    .mode = .mode,
+    ...,
+    .config_path = .config_path,
+    .wait = .wait,
+    .dry_run = .dry_run,
+    .directory = .directory
+  )
+
+  return(res_list)
+}
+
+
+#' Submits multiple NONMEM models in batch via babylon
+#' @param .mods A list of S3 objects of class `bbi_nonmem_model`
+#' @importFrom stringr str_detect
+#' @importFrom tools file_path_sans_ext
+#' @importFrom purrr map
+#' @return A list of S3 objects of class `babylon_process`
+#' @rdname submit_models
+submit_nonmem_models <- function(.mods,
+                                .bbi_args = NULL,
+                                .mode = c("sge", "local"),
+                                ...,
+                                .config_path=NULL,
+                                .wait = TRUE,
+                                .dry_run=FALSE) {
+
+  # check against YAML
+  for (.mod in .mods) { check_yaml_in_sync(.mod) }
+
+  # check for valid type arg
+  .mode <- match.arg(.mode)
+
+  # build command line args
+  param_list <- build_bbi_param_list(.mods, .bbi_args)
+  ###### throw in a strict_mode_error if too many sets? !!!!!!
+
+  # build command line args
+  cmd_args_list <- map(param_list, function(.run) {
+    cmd_args <- c("nonmem", "run", .mode, .run[[YAML_MOD_PATH]], .run[[YAML_BBI_ARGS]])
+
+    # add config path
+    if (!is.null(.config_path)) {
+      cmd_args <- c(cmd_args, sprintf("--config=%s", .config_path))
+    }
+
+   list(
+        cmd_args = cmd_args,
+        model_dir = .run[[WORKING_DIR]]
+      )
+  })
+
+  if (.dry_run) {
+    # construct fake res object
+    return(map(
+      cmd_args_list,
+      function(.run) { bbi_dry_run(.run$cmd_args, .run$model_dir) }
+      ))
+  }
+
+  # launch models
+  res_list <- map(
+    cmd_args_list,
+    function(.run, ...) { bbi_exec(.run$cmd_args, .wait = .wait, .dir = .run$model_dir, ...) }
```

Comment:
no need to declare this function with the ..., those dots actually are going to attempt to inherit anyway from the ... in the map call.